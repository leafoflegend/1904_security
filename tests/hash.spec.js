const hash = require('../server/database/utils/hash.js');

describe('Hash Function', () => {
  it('Given a string, returns a string.', () => {
    expect(typeof hash('alaksndlaknsdlk')).toEqual('string');
  });

  it('Given the same string twice, returns the same string twice.', () => {
    const stringToHash = 'hash me';

    const firstResult = hash(stringToHash);
    const secondResult = hash(stringToHash);

    expect(firstResult).toEqual(secondResult);
  });

  it('Given a string, hashes the string.', () => {
    const hashString = hash('password');

    expect(hashString).toMatchSnapshot();
  });
});
