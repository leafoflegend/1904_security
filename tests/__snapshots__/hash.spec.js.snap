// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hash Function Given a string, hashes the string. 1`] = `"224194230230238222228200"`;
